src/prism.c

```diff
@@ -6912,7 +6912,7 @@ parser_lex(pm_parser_t *parser) {
                         );
                     }
 
-                    if (lex_state_beg_p(parser) || spcarg) {
+                    if (lex_state_beg_p(parser) || spcarg || parser->in_keyword_arg) {
```

Comment:
I'm nervous about this change because it doesn't match up with `parse.y`, which makes me think we're missing some state change/detail. This line is equivalent to https://github.com/ruby/ruby/blob/486b674e2a8437bacb00c48038c04aec420c47a0/parse.y#L10801 which doesn't check if we're inside a keyword argument.

I did a little debugging and it looks like we're transitioning `BEG -> CMDARG` and they're transitioning `BEG -> ARG`. Maybe we're missing something here?

Reply:
We are transitioning to `CMDARG` here:

https://github.com/ruby/prism/blob/2dc5320289a01508e7fd59cc4ef8434d37706085/src/prism.c#L8840-L8848

because `previous_command_start` is `true`. An it is set to be `true` in https://github.com/ruby/prism/blob/2dc5320289a01508e7fd59cc4ef8434d37706085/src/prism.c#L14345-L14350

Reply:
Yeah I wonder if that's not correct or if we're missing something here. Because we should definitely be transitioning to `ARG` and not `CMDARG`.

Reply:
It seemed quite unlikely but can `lex_state_beg_p` be just wrong? I attempted to fix it here and it seems like this bug gets resolved https://github.com/ruby/prism/pull/1591/commits/e854d479ad11bac055a3f80142b82fbf022771e4 Comparing with the Ruby implementation: https://github.com/ruby/ruby/blob/486b674e2a8437bacb00c48038c04aec420c47a0/parse.y#L347

Reply:
Also seems to fix https://github.com/ruby/prism/issues/1572